Compilation warning Wsign-compare

Very silly, but annoying; I get a:

> thread_pool.hpp:107:19: warning: comparison of integer expressions of different signedness: ‘int’ and ‘unsigned int’ [-Wsign-compare]

warning, which is easily silenced by:

```
- for(int i = 0;i < num_threads;i++){
+ for(unsigned int i = 0;i < num_threads;i++){
```

--- a/thread_pool.hpp
+++ b/thread_pool.hpp
@@ -105,7 +105,7 @@ protected:
 	 */
 	void init_threads(){
 	 	task_mutex.lock();
-		for(int i = 0;i < num_threads;i++){
+		for(unsigned int i = 0;i < num_threads;i++){
 			auto f = std::bind(&base_thread_pool::thread_func, this);
 			threads.push_back(std::move(std::thread(f)));
 	 	}

